### PR TITLE
[CVP-1604] Optimize existing layers in Dockerfile

### PR DESCRIPTION
**Pull request by samvarankashyap**
Adding more changes to optimize the Dockerfile.

### CODE REVIEW

**Comment by nmars** [comment]
lgtm

**Review by nmars — APPROVED** [review]

File: Dockerfiles/Dockerfile [hunk 1]
```diff
@@ -10,9 +10,8 @@ RUN export ARCH=$(case $(arch) in x86_64) echo -n amd64 ;; aarch64) echo -n arm6
     curl -LO ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH};\
     chmod +x operator-sdk_${OS}_${ARCH};\
     mv operator-sdk_${OS}_${ARCH} operator-sdk;\
-    mv operator-sdk /usr/local/bin/;
```
Comment (lslebodn) [inline Dockerfiles/Dockerfile:13]: Use `&&` instead of `;` for chain of commands.

File: Dockerfiles/Dockerfile [hunk 2]
```diff
@@ -10,9 +10,8 @@ RUN export ARCH=$(case $(arch) in x86_64) echo -n amd64 ;; aarch64) echo -n arm6
...
-RUN dnf install -y git ansible wget python3-pip
```
Comment (lslebodn) [inline Dockerfiles/Dockerfile:14]: Please use single command per line

File: Dockerfiles/Dockerfile [hunk 3]
```diff
@@ -10,9 +10,8 @@ RUN export ARCH=$(case $(arch) in x86_64) echo -n amd64 ;; aarch64) echo -n arm6
...
-RUN git clone --branch $OPERATOR_TEST_PLAYBOOKS_TAG https://github.com/redhat-operator-ecosystem/operator-test-playbooks.git
```
Comment (lslebodn) [inline Dockerfiles/Dockerfile:15]: Unnecessary semicolon at the end

File: Dockerfiles/Dockerfile (hunk 2 above)
Comment (lslebodn) [reply Dockerfiles/Dockerfile:14]: Please replace `git` with `git-core` and disable weak dependencies

Lot of unnecessary weak dependencies were installed/
```
Installing weak dependencies:
 perl-IO-Socket-IP        noarch   0.41-2.fc33                  fedora     43 k
 perl-IO-Socket-SSL       noarch   2.068-3.fc33                 fedora    216 k
 perl-Mozilla-CA          noarch   20200520-3.fc33              fedora     12 k
 perl-NDBM_File           x86_64   1.15-467.fc33                updates    33 k
 python3-invoke           noarch   1.4.1-4.fc33                 fedora    149 k
 python3-paramiko         noarch   2.7.2-1.fc33                 fedora    287 k
 python3-pyasn1           noarch   0.4.8-3.fc33                 fedora    133 k
 python3-winrm            noarch   0.3.0-11.fc33                fedora     59 k

```

**Comment by lslebodn** [comment]
Please remove line 
```
RUN rpm -i https://kojipkgs.fedoraproject.org//packages/fedora-repos/33/3/src/fedora-repos-33-3.src.rpm
```

Installing src.rpm is not required (I tested it locally)
I cannot fix anything. src.rpm are just use for rebuilding sources to real rpms.

**Comment by lslebodn** [comment]
What is a purpose of the line `RUN mkdir /project/output`?
And why does it need to be in separate instruction?

**Comment by samvarankashyap** [comment]
> What is a purpose of the line `RUN mkdir /project/output`?
> And why does it need to be in separate instruction?

Need not be. I have fixed it.

File: Dockerfiles/Dockerfile (hunk 1 above)
Comment (samvarankashyap) [reply Dockerfiles/Dockerfile:13]: export COMMAND does not work with && . Rest of the places i have updated required.

File: Dockerfiles/Dockerfile [hunk 4]
```diff
@@ -3,16 +3,15 @@ RUN rpm -i https://kojipkgs.fedoraproject.org//packages/fedora-repos/33/3/src/fe
...
+RUN export ARCH=$(case $(arch) in x86_64) echo -n amd64 ;; aarch64) echo -n arm64 ;; *) echo -n $(arch) ;; esac);\
+    export OS=$(uname | awk '{print tolower($0)}');\
+    export OPERATOR_SDK_DL_URL=https://github.com/operator-framework/operator-sdk/releases/download/$OPERATOR_SDK_VERSION/;\
+    curl -LO ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH} && \
+    chmod +x operator-sdk_${OS}_${ARCH} && \
+    mv operator-sdk_${OS}_${ARCH} operator-sdk && \
+    mv operator-sdk /usr/local/bin/ && \
```
Comment (lslebodn) [inline Dockerfiles/Dockerfile:12]: Two move command are bit complicated we can simplify it with storing file the the expected location in curl.

Simpler code is easier to understand
```
curl -L -o /usr/local/bin/operator-sdk ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH} \
&& chmod a+x  /usr/local/bin/operator-sdk
```

File: Dockerfiles/Dockerfile (hunk 1 above)
Comment (lslebodn) [reply Dockerfiles/Dockerfile:13]: because export is not a command :-) and it is fine to use it there

File: Dockerfiles/Dockerfile (hunk 2 above)
Comment (lslebodn) [reply Dockerfiles/Dockerfile:14]: This was not resolved

Comment (samvarankashyap) [reply Dockerfiles/Dockerfile:14]: according to the changelog its resolved 
@lslebodn

Comment (samvarankashyap) [reply Dockerfiles/Dockerfile:14]: @lslebodn Do you mean the weak dependencies?  I will fix it

File: Dockerfiles/Dockerfile (hunk 4 above)
Comment (samvarankashyap) [reply Dockerfiles/Dockerfile:12]: +1

File: Dockerfiles/Dockerfile (hunk 2 above)
Comment (lslebodn) [reply Dockerfiles/Dockerfile:14]: I request two things in the thread.

File: Dockerfiles/Dockerfile [hunk 5]
```diff
@@ -3,16 +3,13 @@ RUN rpm -i https://kojipkgs.fedoraproject.org//packages/fedora-repos/33/3/src/fe
...
-    mv operator-sdk /usr/local/bin/;
-RUN dnf install -y git ansible wget python3-pip
-RUN git clone --branch $OPERATOR_TEST_PLAYBOOKS_TAG https://github.com/redhat-operator-ecosystem/operator-test-playbooks.git
+RUN export ARCH=$(case $(arch) in x86_64) echo -n amd64 ;; aarch64) echo -n arm64 ;; *) echo -n $(arch) ;; esac);\
+    export OS=$(uname | awk '{print tolower($0)}');\
+    export OPERATOR_SDK_DL_URL=https://github.com/operator-framework/operator-sdk/releases/download/$OPERATOR_SDK_VERSION/;\
+    curl -L -o /usr/local/bin/operator-sdk ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH} \ 
+    && chmod a+x /usr/local/bin/operator-sdk && \
+    mkdir /project/output && \
+    dnf install --setopt=install_weak_deps=False -y git-core ansible && dnf clean all && \
```
Comment (lslebodn) [inline Dockerfiles/Dockerfile:12]: there are still two commands on the same line

**Comment by samvarankashyap** [comment]
> Please remove line
> 
> ```
> RUN rpm -i https://kojipkgs.fedoraproject.org//packages/fedora-repos/33/3/src/fedora-repos-33-3.src.rpm
> ```
> 
> Installing src.rpm is not required (I tested it locally)
> I cannot fix anything. src.rpm are just use for rebuilding sources to real rpms.

Removing the sprm

File: Dockerfiles/Dockerfile (hunk 5 above)
Comment (samvarankashyap) [reply Dockerfiles/Dockerfile]: I have broken them into two separate lines

File: Dockerfiles/Dockerfile (hunk 2 above)
Comment (samvarankashyap) [reply Dockerfiles/Dockerfile:14]: updated to git-core and added --setopt=install_weak_deps=False

**Review by lslebodn — APPROVED** [review]
+1

**Pull request merged**